### DISCUSSION TITLE
Field Aliases

### DISCUSSION THREAD

**Discussion by malasee** [byme8/ZeroQL#107] [Q&A]
Good afternoon

I have a requirement where I'm calling the same query field with different arguments within the same query.  Here is a simplified example of what I'm trying to achieve:

```csharp
        var user1Id = "user1";
        var user2Id = "user2";
        
        var response = await zeroQLClient.Query(o => new
        {
            User1 = o.UserById(user1Id, u => new { u.Id, u.FirstName, u.LastName }),
            User2 = o.UserById(user2Id, u => new { u.Id, u.FirstName, u.LastName })
        });
```

This results in an error "Fields userById conflicts because they have differing arguments. Use different aliases on the fields to fetch both if this was intentional."

Is there any way I can provide an alias for each of these?  Is there another approach to achieve this?

Thanks

**Comment by byme8** [comment]
At the moment, there is no way how to call two identical fields twice in one request. I have ideas how to make it work but, at the moment, there is no an easy way to support it.

**Comment by mpal035** [comment]
Is this also the case for Mutations with sub mutations? Is this also not supported yet
